src/UtilityRepository.js

```diff
@@ -0,0 +1,42 @@
+class UtilityRepository {
+  constructor(userData, sleepData, hydrationData, activityData) {
+    this.userData = userData;
+    this.sleepData = sleepData;
+    this.hydrationData = hydrationData;
+    this.activityData = activityData;
+  }
+  
+  getUserLogs(id, dataType) {
+    return this[dataType].filter(user => user.userID === id);
+  }
+
+  getUserInfoByDate(id, date, dataType) {
+    return this.getUserLogs(id, dataType).find(log => log.date === date);
+  }
+
+  getAllUserInfoByDate(date, dataType) {
+    return this[dataType].filter(log => log.date === date);
+  }
+
+  getWeeksData(id, date, dataType) {
+    const userInfo = this.getUserLogs(id, dataType);
+    let i = userInfo.findIndex(day => day.date === date);
+    return userInfo.slice(i - 6, i + 1);
+  }
+
+  getTotal(id, dataType, dataKey) {
+    const userInfo = this.getUserLogs(id, dataType);
```

Comment:
Very interesting approach in returning an object, do you think you could elaborate on why the object returns the `total` and `userInfo`?

Reply:
When you put a variable name in an object that's been assigned already you can make the variable name the key and the value of the variable the value of the key in the object.  
Ex. const bug = 'insect'; { bug } is the same as { bug: 'insect' }.